ibis/expr/types/relations.py

```diff
@@ -1722,30 +1722,58 @@ def select(
 
     projection = select
 
+    @util.deprecated(
+        as_of="7.0",
+        instead=(
+            "use `Table.rename` instead (if passing a mapping, note the meaning "
+            "of keys and values are swapped in Table.rename)."
+        ),
+    )
     def relabel(
         self,
         substitutions: Mapping[str, str]
         | Callable[[str], str | None]
         | str
         | Literal["snake_case", "ALL_CAPS"],
+    ) -> Table:
+        """Deprecated in favor of `Table.rename`"""
+        if isinstance(substitutions, Mapping):
+            substitutions = {new: old for old, new in substitutions.items()}
+        return self.rename(substitutions)
+
+    def rename(
+        self,
+        method: str
+        | Callable[[str], str | None]
```

Comment:
document what returning `None` from the lambda does. I think it results in no renaming? I would prefer if it were simpler: you always had to return a str, and if you wanted no renaming, then just return the original string.

Reply:
Sure, I'll update the docstring. This is existing behavior in `relabel`, I'd be reluctant to change it.

> if you wanted no renaming, then just return the original string.

The existence of support for `None` doesn't prevent you from doing this. If this is how you'd want to use this API, then that should work for you.

Reply:
ah, the existing behavior probably is the decider for this.